### PR TITLE
Explicitly use hex encoding in GUI to RPC API

### PR DESCRIPTION
**Pull request by JeremyRand**
Prevents corruption issues that were happening when default encoding wasn't ASCII.  Also a prereq to supporting binary data in the GUI.

### CODE REVIEW

**Comment by JeremyRand** [comment]
@domob1812 This PR is ready for review.

**Review by domob1812 — CHANGES_REQUESTED** [review]
Looks good, but please consider the comments I've added.  Also I think the commit history should be squashed - all into one commit seems appropriate here to me.

File: src/qt/buynamespage.cpp
Comment (JeremyRand) [reply src/qt/buynamespage.cpp:92]: Seems this code has the same type-safety issue that you brought up in https://github.com/namecoin/namecoin-core/pull/482#discussion_r776625657 .  Would it be okay if I refactored this so that the string argument is in hex form rather than binary form?

Comment (domob1812) [reply src/qt/buynamespage.cpp:92]: Yes sure.

Comment (JeremyRand) [reply src/qt/buynamespage.cpp]: I decided to defer this refactor until the GUI is updated to handle hex data, and instead just added a conversion function as you requested.

File: src/qt/nametablemodel.cpp
Comment (JeremyRand) [reply src/qt/nametablemodel.cpp:76]: Not yet; this PR uses hex encoding for talking to the RPC API but converts back to ASCII, so encoding errors will still be an issue for now.

**Comment by JeremyRand** [comment]
@domob1812 Addressed review, and fixed merge conflict via rebase.  Feel free to proceed with review; I'll squash before you merge.

File: src/qt/nametablemodel.cpp
Comment (domob1812) [reply src/qt/nametablemodel.cpp:76]: Ok, but then I think it should be reformulated.  As it is currently, it makes no sense - the TODO says "set encodings to hex", but you actually do that in the very next line.

**Review by domob1812 — APPROVED** [review]
Thanks for the update, looks good.  Please just consider updating the TODO text, otherwise this is good for me.

File: src/qt/nametablemodel.cpp
Comment (JeremyRand) [reply src/qt/nametablemodel.cpp]: Done.

**Comment by JeremyRand** [comment]
@domob1812 Review addressed; let me know if this is ready for a squash.

File: src/qt/nametablemodel.cpp
Comment (domob1812) [inline src/qt/nametablemodel.cpp:131]: Nit:  I think it would be more readable if you merged the two `try-catch` into one, which has the same effect in the end.

File: src/qt/nametablemodel.cpp
Comment (domob1812) [inline src/qt/nametablemodel.cpp:201]: Same here of course.

**Review by domob1812 — APPROVED** [review]
Thanks for the updates.  It looks good now, please just consider merging the two `try-catch` blocks in both places.  Then it is good to merge for me.

**Comment by JeremyRand** [comment]
@domob1812 Review addressed once again; let me know if this is ready for a squash.

**Review by domob1812 — APPROVED** [review]
ACK, I think this is ready for squashing and merging.

**Comment by JeremyRand** [comment]
Rebased to fix merge conflict. I still need to confirm that the rebase didn't break anything, will test that in the next day or two.

**Comment by JeremyRand** [comment]
Blocked by https://github.com/namecoin/namecoin-core/pull/536

**Comment by JeremyRand** [comment]
> Blocked by #536

#536 is merged. @domob1812 this is ready for review.

**Comment by domob1812** [comment]
All looks good, thanks!  ACK 922da75795fed146462184468447d8bc2167f19d.

**Comment by victoirechan** [comment]
At 2024-10-16 00:42:52, "Daniel Kraft" ***@***.***> wrote:

All looks good, thanks! ACK 922da75.

**Pull request merged**
